src/components/BraintreePayPalButton.test.js

```diff
@@ -0,0 +1,228 @@
+
+const setup = () => {
+    document.body.innerHTML = "";
+
+    window.paypal = {
+        Buttons: jest.fn(() => ({
+            close: jest.fn().mockResolvedValue(),
+            isEligible: jest.fn().mockReturnValue(true),
+            render: jest.fn().mockResolvedValue({}),
+        })),
+    };
+    window.braintree = {
+        client: {
+            create: jest.fn(),
+        },
+        paypalCheckout: {
+            create: jest.fn().mockReturnValue({
+                create: jest.fn(),
+                tokenizePayment: jest.fn(),
+                createPayment: jest.fn(),
+            }),
+        },
+    };
+
+    loadScript.mockResolvedValue(window.paypal);
+    loadCustomScript.mockResolvedValue(window.braintree);
+};
+
+describe("Braintree PayPal button fail in mount process", () => {
+    beforeAll(setup);
```

Comment:
Let's use beforeEach for a clean state for each test
```suggestion
    beforeEach(setup);
```